lib/cloudregister/registerutils.py

```diff
@@ -101,13 +107,21 @@ def clean_hosts_file(domain_name):
         content = hosts_file.readlines()
 
     smt_announce_found = None
+    smt_domain_found = None
     for entry in content:
         if b'# Added by SMT' in entry:
             smt_announce_found = True
             continue
         if smt_announce_found and domain_name in entry:
             smt_announce_found = False
+            smt_domain_found = True
             continue
+        if smt_domain_found:
+            smt_domain_found = False
+            if b'registry' in entry and domain_name in entry:
+                # registry entry is optional
+                continue
+
```

Comment:
In the diff it looks like the `new_hosts_content.append(entry)` moved outside the for loop, please double check this is not the case.

Reply:
I checked, it is inside the for loop in the code